subsys/location/Kconfig

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2022, Bjarki Arge Andreasen
+# SPDX-License-Identifier: Apache-2.0
+
+config LOCATION
+	bool "Location subsystem"
+	help
+	  Include location subsystem
+
+if LOCATION
+
+config LOCATION_PROVIDERS_MAX
+	int "Max supported providers"
+	default 4
+	help
+	  Sets the maximum supported providers
```

Comment:
I wonder if we could avoid specifying the max providers by utilizing the `STRUCT_SECTION_ITERABLE` macro that could add the providers into a special linker section and then we could just iterate through that section at runtime. This pattern is used in many places in zephyr and it avoids allocating any extra data that might not be used.

Reply:
The latest commit only changes 6 files, and does not include the subsystem at all :) That is from the previous design.

Reply:
I have squashed the two commits so it it clearer :)